test/jasmine/tests/modebar_test.js

```diff
@@ -288,9 +294,15 @@ describe('ModeBar', function() {
         it('removes the mode bar entirely', function() {
             var modeBarParent = modeBar.element.parentNode;
 
+            var style = document.querySelector('style[id*="modebar"]');
+            expect(style).toBeTruthy();
+
             modeBar.destroy();
 
             expect(modeBarParent.querySelector('.modebar')).toBeNull();
+
+            style = document.querySelector('style[id*="modebar"]');
+            expect(style).toBeNull();
```

Comment:
This is failing on the CI even though it's working on my Windows 10 laptop and my Linux workstation at home 🤷‍♂️. Do you have any idea why?

Reply:
`npm run test-jasmine -- modebar` works fine but

`npm run test-jasmine -- --nowatch` does not.

Something from another suite (before `modebar_test.js` alphabetically) must be not get cleared properly.

Reply:
Ok, hopefully I will find the misbehaving tests with this command:
`find test/jasmine/tests/ -type f -printf "%f\n" | xargs -I {} xvfb-run -a npm run test-jasmine -- {} modebar_test.js --nowatch`

Reply:
Ok I found a few. My intuition tells me the problem is that they solely use `destroyGraphDiv` to clear the stage which is not good enough sometimes. For example, responsive plots attach event listeners on the window, `style` elements are injected into the page on load, and so on. `Plotly.purge` should probably be used instead. I will confirm tomorrow that this is indeed the case.

Reply:
Might be best to just pre-clean for this suite, rather than try and sanitize all the other tests. So something up at the top of the suite like
```js
beforeAll(function() {
    d3.selectAll('style[id*="modebar"]').remove();
});
```

Reply:
Or you could put `d3.selectAll('style[id*="modebar"]').remove();` into `destroyGraphDiv`. That routine is wholly predicated on the idea that we only have one plot at a time in the tests, so there should be no problem deleting style sheets added by a different plot. Possibly a whole `Plotly.purge(gd)` could go into `destroyGraphDiv` but that might have been omitted for performance reasons...

Reply:
I decided to make my query selector more explicit to target the current figure's modebar. That way, I didn't have to touch the other tests.

I also feel like a `Plotly.purge(gd)` could/should go into `destroyGraphDiv` but that seemed like a pretty big change just to get this PR through.